File: reference/dotnet/Domain/Company.Product.Domain.UseCases/Types/Cat.cs
using System;

namespace Company.Product.Domain.UseCases.Types
{
    public class Cat : Animal
    {
        public string B { get; }
     
        public Cat(Guid animalId, string sound, string b) 
            : base(animalId, sound)
        {
            B = b;
        }

        public override T Visit<T>(AnimalVisitor<T> visitor)
        {
            if (visitor is null)
            {
                throw new ArgumentNullException(nameof(visitor));
            }

            return visitor.VisitCat(this);
        }
    }
}

File: reference/springboot/domain-usecases/src/main/java/org/example/domain/usecases/types/Cat.java
package org.example.domain.usecases.types;

import lombok.Builder;
import lombok.Getter;

import java.util.UUID;

@Getter
public class Cat extends Animal {
    private String b;

    @Builder
    public Cat(UUID id, String sound, String b) {
        super(id, sound);
        this.b = b;
    }

    @Override
    public <T> T visit(AnimalVisitor<T> visitor) {
        return visitor.visitCat(this);
    }
}
